src/hiccup/util.clj

```diff
@@ -123,3 +124,16 @@
             (str "?" (url-encode params))
             params)))))
 
+(defn fast-run
+  [callback coll]
+  (when coll
+    (let [^java.util.Iterator iterator (.iterator ^Iterable coll)]
+      (while (.hasNext iterator)
+        (callback (.next iterator))))))
+
+(defmacro build-string
+  [& strs]
+  (let [w (gensym)]
+    `(let [~(vary-meta w assoc :tag `java.lang.StringBuilder) (StringBuilder.)]
+       ~@(map (fn [arg] `(.append ~w ~arg)) strs)
+       (.toString ~w))))
```

Comment:
How does `build-string` compare to `str` in terms of performance?

Reply:
clojure.core/str is a variadic function: **clojure.core/str [x & ys]**

It's a popular helper to avoid it. There is a Clojure library for this, for instance: https://github.com/kumarshantanu/stringer

Reply:
Hmm, indeed, the reflection is not happening. We can remove it. We can remove it

Reply:
8bc343fa9dccdec2db0ecb83cbfc4228f76f155b